Tests/Get-JenkinsUserFullName.Test.ps1

```diff
@@ -0,0 +1,110 @@
+Import-Module -Name (Get-ChildItem $PSScriptRoot\..\*.psm1 | Select-Object -first 1).FullName -Force
+. "$PSScriptRoot\..\Public\Get-JenkinsUserFullName.ps1"
+. "$PSScriptRoot\..\Private\Get-JenkinsUserInfo.ps1"
+
+Describe 'Get-JenkinsUserFullName method' {
+
+    AfterEach {
+        $script:AllUserInfo = $null
+    }
+
+    $Fullname1 = "Robot"
+    $UserInfo1 = @"
+{
+        "_class": "hudson.model.User",
+        "absoluteUrl": "https://localhost/jenkins/user/nothuman",
+        "description": null,
+        "fullName": "$Fullname1",
+        "id": "nothuman",
+        "property": [
+            {
+                "_class": "hudson.tasks.Mailer`$UserProperty",
+                "address": "nothuman@localhost"
+            }
+        ]
+}
+"@ | ConvertFrom-Json
+
+    $Fullname2 = "praise the sun"
+    $UserInfo2 = @"
+{
+        "_class": "hudson.model.User",
+        "absoluteUrl": "https://localhost/jenkins/user/delsol",
+        "description": null,
+        "fullName": "$Fullname2",
+        "id": "delsol",
+        "property": [
+            {
+                "_class": "hudson.tasks.Mailer`$UserProperty",
+                "address": "delsol@localhost"
+            }
+        ]
+}
+"@ | ConvertFrom-Json
+
+
+    Context 'When a fullname is initially requested' {
+
+        It 'Retrieves the fullname of the user' {
+            $user = $UserInfo1.id
+            $pass = "changeme"  | ConvertTo-SecureString -AsPlainText -Force
+
+            Mock -CommandName Get-JenkinsUserInfo { return $UserInfo1 } -Verifiable
+            $fullName = Get-JenkinsUserFullName -UserName $user -Password $pass
+            $fullName | Should -BeExactly $Fullname1
+        }
+
+        It 'It is looked up, cached and re-used' {
+            $user = $UserInfo2.id
+            $pass = "changeme"  | ConvertTo-SecureString -AsPlainText -Force
+
+            Mock -CommandName Get-JenkinsUserInfo { return $UserInfo2 } -Verifiable
+            $name = Get-JenkinsUserFullName -UserName $user -Password $pass
+            $name | Should -BeExactly $Fullname2
+        }
+    }
+
+    Context 'When fullname is requested at least twice for the same user, against itself' {
+
+        It '1st user against itself' {
```

Comment:
Perhaps this description could be clearer, like "It returns the same content twice," or something along those lines?

Reply:
Updated to reflect more information in the test description.